app/models/user.rb

```diff
@@ -1168,12 +1168,16 @@ def self.trainee_delegate_ids
   end
 
   def self.search(query, params: {})
-    users = Person.includes(:user).current
-    # We can't search by email on the 'Person' table
-    search_by_email = false
-    unless ActiveRecord::Type::Boolean.new.cast(params[:persons_table])
+    search_by_email = ActiveRecord::Type::Boolean.new.cast(params[:email])
+    admin_search = ActiveRecord::Type::Boolean.new.cast(params[:adminSearch])
+
+    return User.where(email: query) if admin_search && search_by_email
+
+    if ActiveRecord::Type::Boolean.new.cast(params[:persons_table])
```

Comment:
This thing within the `if` can probably be converted to its own `searching_persons_table` boolean variable. That same variable can then be used above already as `search_by_email = searching_persons_table && ActiveRecord::Type::Boolean.new.cast(...)`

Reply:
Changed to own boolean as you suggested, also noted the sidenote.